addons/website/models/website_page.py

```diff
@@ -229,6 +238,94 @@ def _set_cache_response(self, cache_key, response):
         """ Put in cache the given response. """
         self.pool._Registry__cache[('website.page', _cached_response, self.id, cache_key)] = response
 
+    @api.model
+    def _search_get_detail(self, website, order, options):
+        with_description = options['displayDescription']
+        requires_sudo = False
+        domain = [website.website_domain()]
+        if not self.env.user.has_group('website.group_website_designer'):
+            domain.append([('website_published', '=', True)])
```

Comment:
If this domain is then only used through ORM search methods (so, not injected into handcrafted SQL), it shouldn't be needed as it is automatically added with an `ir.rule`.

```xml
<record id="website_page_rule_public" model="ir.rule">
        <field name="name">website.page: portal/public: read published pages</field>
        <field name="model_id" ref="website.model_website_page"/>
        <field name="domain_force">[('website_published', '=', True)]</field>
        <field name="groups" eval="[(4, ref('base.group_portal')), (4, ref('base.group_public'))]"/>
    </record>
```

Reply:
But rule is not applied because of sudo.